format/README.md

```diff
@@ -0,0 +1,25 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Arrow Protocol Files
+
+This folder contains binary protocol definitions for the Arrow columnar format
+and other parts of the project, like the Flight RPC framework.
+
+This is a copy of https://github.com/apache/arrow/tree/main/format .
```

Comment:
Would it make sense for these format files to live in a new `arrow-format` repository, so they can be included as a submodule and kept up to date in one place rather than need to be manually updated?

I briefly looked at other Arrow implementation repositories (go, swift, js, rust) and as far as I can tell only the Rust implementation also has a copy of these files. I guess this is because the other implementations have checked in the generated code, or they use hand-written implementations rather than generated code.

These files probably won't change very often though so I think it's fine to store copies here and manually update files as needed. And if we did want to change this, it should be a follow-up change and not block this PR.